install/traf_checkset_env.sh

```diff
@@ -0,0 +1,186 @@
+http://download.icu-project.org/files/icu4c/4.4/icu4c-4_4-src.tgz
+https://archive.apache.org/dist/zookeeper/zookeeper-3.4.5/zookeeper-3.4.5.tar.gz
+http://archive.apache.org/dist/thrift/0.9.0/thrift-0.9.0.tar.gz
+http://archive.apache.org/dist/maven/maven-3/3.3.3/binaries/apache-maven-3.3.3-bin.tar.gz
+https://github.com/google/protobuf/releases/download/v2.5.0/protobuf-2.5.0.tar.gz
+https://dist.apache.org/repos/dist/release/logging/log4cxx/0.10.0/apache-log4cxx-0.10.0.tar.gz
+http://archive.apache.org/dist/hadoop/common/hadoop-2.6.0/hadoop-2.6.0.tar.gz
+)
+
+# check the local software directory
+if [ ! -d ${MY_LOCAL_SW_DIST} ]; then
+    echo "WARN: local sofrware tools aren't exist, should be downloading all
+    software tools from internet, it's very slowlly! While you can
+    prepare local software, and set environment MY_LOCAL_SW_DIST to
+    change default software path, and run this script again. The 
+    default local software directory is [${MY_LOCAL_SW_DIST}]. You
+    can set environment MY_LOCAL_SW_DIST to change it."
+    echo "Do you want continue? Enter y/n (default:n):"
+    read YN
+    case ${YN} in
+	Y|y|Yes|YES)
+	    ;;
+	*)
+	    echo "Download following build tools from internet:"
+	    for i in `seq ${#local_sws[@]}`
+	    do
+		printf "%2d.%15s: %s\n" ${i} ${local_sws[${i}-1]} ${http_sws[${i}-1]}
+	    done
+	    exit 1
+	    ;;
+    esac
+else
+    # check the local software's source exist or not
+    for local_sw in ${local_sws[@]}
+    do
+	local_file=`ls ${MY_LOCAL_SW_DIST} | grep ${local_sw}`
+	if [ "x${local_file}" = "x" ]; then
+            echo "WARN: [${local_sw}] source file not exist in directory [${MY_LOCAL_SW_DIST}]"
+	fi
+    done
+fi
+
+# check the permission.
+(${MY_SUDO} touch /etc/trafpermise) >>${LOGFILE}2>&1
+if [ "x$?" != "x0" ]; then
+    echo "ERROR: you must run this script with sudo without password permission."
+    exit 1
+fi
+
+# check the based command
+basecmds=(yum lsb_release awk cut uname)
+for basecmd in ${basecmds[@]}  
+do
+    basecmdpath=`which ${basecmd}`
+    if [ "x$?" != "x0" ]; then
+	case ${basecmd} in
+	    yum)
+		echo "ERROR: yum must install first by yourself."
+		exit 1
+		;;
+	    lsb_release)
+		(${MY_YUM} install redhat-lsb) >>${LOGFILE}2>&1
+		if [ "x$?" = "x0" ]; then
+		    echo "ERROR: yum repo server has a error when run command [${MY_YUM} install redhat-lsb]."
+		    exit 1
+		fi
+		;;
+	    *)
+		echo "ERROR: command [${basecmd}] not exist, make sure you have installed,and the path command added to path rightly."
+		exit 1
+	esac
+    fi
+    echo "INFO: command ${basecmd} exist"
+done
+
+osname=`uname -o`
+cpuarch=`uname -m`
+echo "INFO: os is [${osname}], cpu architecture is [${cpuarch}]"
+
+osdistributor=`lsb_release -i | cut -d : -f 2-`
+case ${osdistributor} in
+    *RedHat*|*CentOS*)
+	echo "INFO: os distributor id is [${osdistributor}]"
+	;;
+    *)
+	echo "ERROR: os distributor is [${osdistributor}], but only support RedHat or CentOS now."
```

Comment:
Suggest: "ERROR: The OS distribution is [${osdistributor}], but Trafodion only supports RedHat and CentOS presently."